spec/core/dataset_spec.rb

```diff
@@ -2279,6 +2279,14 @@ def ==(o) @h == o.h; end
     db.sqls.must_equal ["SELECT 1 AS one FROM test WHERE 'f' LIMIT 1"]
   end
 
+  it "should return false if dataset contains record with null field" do
```

Comment:
The issue you are trying to fix is only present for datasets using custom SQL (no SQL server will return a NULL value for SELECT 1).  So the spec should also be modified to use custom SQL.

Reply:
I'll need help with modifying the spec to use custom SQL.

Also, we will need tests for `DB.values([[nil, 1], [2, 3]]).empty?`, any other non-custom SQL cases that also need to be worked around, and maybe the loading a large amount of data case. I'll need help with these.